smarts/env/hiway_env.py

```diff
@@ -164,7 +165,11 @@ def step(self, agent_actions):
             agent_id: self._agent_specs[agent_id].action_adapter(action)
             for agent_id, action in agent_actions.items()
         }
-        observations, rewards, agent_dones, extras = self._smarts.step(agent_actions)
+        observations, rewards, agent_dones, extras = None, None, None, None
+        with timeit("SMARTS Simulation/Scenario Step", self._log):
+            observations, rewards, agent_dones, extras = self._smarts.step(
+                agent_actions
+            )
```

Comment:
I don't really have a huge problem with all of these (and I think I originally suggested the issue), but I'm also not a huge fan of what this does to our code readability.

Another option throughout would be to ensure that our standard log messages all include a timestamp (e.g., by using the `format` param to `logging.basicConfig()`) and then just add more log messages before and after operations like:
    self._log.info("stepping all providers...")
    self._provider_step()
    self._log.info("doing the next thing...")
    self._next_thing()
    self._log.info("etc.") 
These often can serve as (and replace) code comments too.

Anyway, I'm not saying that's necessarily better (b/c they require you to actually do math when you look at the log file to see how long things took), but at least it doesn't change variable scoping.

Reply:
I thought the idea here was to precisely measure how long each step will take here which the current changes allow use to by wrapping those steps in a `contextmanager`. The changes you are proposing may improve code readability but won't give precise measuring since each step takes milliseconds to finish.

Reply:
Well, if we want precision, we can profile it (and/or use flamegraph).  I originally suggested this issue (to Tucker, who then filed it) as a sort of heuristic "poor-man's profiling" to allow us to see delays in the logs without actually going all the way to profiling.  So it wasn't meant for precision originally, although it's not a bad idea.

I could go either way on this PR -- I certainly won't block it.  Maybe let's see what others think.

Reply:
I just changed the logging to your suggestion above since it makes sense not to change variable scoping.